src/custom-elements/src/components/button/button.tsx

```diff
@@ -0,0 +1,108 @@
+import { Component, Event, EventEmitter, Prop, Watch } from '@stencil/core'
+
+@Component({
+  tag: 'chi-button',
+  styleUrl: 'button.scss',
+  scoped: true
+})
+export class Button {
+  /**
+   *  to set button type { float, pill, close, icon }.
+   */
+  @Prop({ reflectToAttr: true }) type: string;
+
+  /**
+   *  to set button color { primary, secondary, danger, dark }.
+   */
+  @Prop({ reflectToAttr: true }) color: string;
+
+  /**
+   *  to set button size { sm, md, lg, xl }.
+   */
+  @Prop({ reflectToAttr: true }) size = 'md';
+
+  /**
+   *  to remove a buttons solid background and keep its colored border.
+   */
+  @Prop({ reflectToAttr: true }) outline = false;
+
+  /**
+   *  to render a button with a transparent background and border.
+   */
+  @Prop({ reflectToAttr: true }) flat = false;
+
+  /**
+   *  to render a button to fill the parent space.
+   */
+  @Prop({ reflectToAttr: true }) fluid = false;
+
+  /**
+   *  to add additional classes to the chi-button.
+   */
+  @Prop({ reflectToAttr: true }) extraClass: string;
```

Comment:
Please, remove this property. Extra classes, like [utility classes](https://assets.ctl.io/chi/1.1.1/utilities/border/), should be applied to the web component itself, not the inner button. So, this property might be confusing.

Reply:
> Please, remove this property. Extra classes, like [utility classes](https://assets.ctl.io/chi/1.1.1/utilities/border/), should be applied to the web component itself, not the inner button. So, this property might be confusing.

@srinath-koneru Review this comment, please.